bundles/mapping/mapwfs2/plugin/WfsVectorLayerPlugin.ol.js

```diff
@@ -380,14 +380,16 @@ export class WfsVectorLayerPlugin extends AbstractMapLayerPlugin {
         }
         Oskari.getSandbox().notifyAll(builder.apply(null, args));
     }
-    saveUserStyle (layer, styleId) {
-        const oskariStyle = this.visualizationForm.getOskariStyle();
-        const name = this.visualizationForm.getOskariStyleName();
-        oskariStyle.id = styleId;
-        const styleWithMetadata = {
-            name: name,
-            style: oskariStyle
-        };
+    saveUserStyle (layer, id) {
+        const style = this.visualizationForm.getOskariStyle();
+        const layerId = layer.getId();
+        let name = this.visualizationForm.getOskariStyleName();
+        if (!name) {
+            const existing = this.userStyleService.getUserStylesForLayer(layerId);
```

Comment:
If this would always return an array the line for count could always be  existing.length + 1